src/GitLabHealth-Model-Analysis/GitMetricExporter.class.st

```diff
@@ -1,604 +1,611 @@
+			  userAnalysisReport metrics at: newMetric name put: result ].
+
+		  userAnalysisReport ]
+]
+
+{ #category : #utilities }
+GitMetricExporter >> getAnalysisWithDate: date from: analysisCollection [
+
+	^analysisCollection
+		detect: [ :groupAnalysis | (groupAnalysis period at: #since) = date ]
+		ifNone: [ nil ]
+]
+
+{ #category : #accessing }
+GitMetricExporter >> glhImporter: anImporter [
+
+	glhImporter := anImporter withInitialCommits: false; yourself. 
+]
+
+{ #category : #adding }
+GitMetricExporter >> importAnalysesFromS3: bucketName accessKey: accessKey secretKey: secretKey region: region [
+
+	| s3 bucket xmlObjects contentElements filesContent |
+	AWSS3Config default
+		accessKeyId: accessKey;
+		secretKey: secretKey;
+		regionName: region.
+
+	s3 := AWSS3 new.
+	bucket := s3 bucketNamed: bucketName.
+	xmlObjects := bucket listObjects.
+	contentElements := xmlObjects root elementsSelect: [ :element |
+		                   element isNamed: 'Contents' ].
+
+	filesContent := contentElements collect: [ :contentElement |
+		                | keyValue |
+		                keyValue := contentElement contentStringAt: 'Key'.
+		                bucket getObject: keyValue ].
+
+	^ analyses := (filesContent collect: [ :file | STON fromString: file ])
+		              flattened
+]
+
+{ #category : #initialization }
+GitMetricExporter >> initialize [
+
+	entities := OrderedCollection new.
+	"set up the minimal date from which we are looking for the commits a particular user in projects"
+	sinceTimeLimit := '2024-01-01' asDate asDateAndTime.
+	runningPeriods := OrderedCollection new.
+	maxCommitWindow := 3.
+	over := Date.
+
+	analyses := OrderedCollection new.
+
+	"User Metrics"
+	metrics := {
+		           CodeAdditionMetric.
+		           CodeDeletionMetric.
+		           CommentContributionMetric.
+		           CommitFrequencyMetric.
+		           CodeChurnMetric.
+		           DelayUntilFirstChurnMetric.
+		           TimeBetweenCommitMetric.
+		           ContributedProjectMetric.
+		           CodeAdditionByMRMetric.
+		           CodeDeletionByMRMetric.
+		           MergeRequestDurationMetric.
+		           OpenedMergeRequestMetric.
+		           ClosedMergeRequestMetric.
+		           MergedMergeRequestMetric.
+		           SelfMergedUnderAMinuteMergeRequestMetric.
+		           ClosedTicketsMetric.
+		           JiraTimeMRTimeDifferenceMetric.
+		           CommitsNumberByJiraTicketsMetric.
+		           CommitsLinkWithTicketMetric.
+
+
+		           "Project Metrics"
+		           CommitFrequencyProjectMetric.
+		           ClosedMergeRequestProjectMetric.
+		           CreatedMergeRequestProjectMetric.
+		           MergedMergeRequestProjectMetric.
+
+		           ClosedMergeRequestDurationProjectMetric.
+		           OpenedMergeRequestDurationProjectMetric.
+		           MergedMergeRequestDurationProjectMetric.
+		           SelfMergedUnderAMinuteMergeRequestProjectMetric.
+               AutoCodeReviewCommitFrequencyProjectMetric

```

Comment:
metrics are not added here. Maybe we should by default specify the script to load every subclass of ProjectMetrics